### DISCUSSION TITLE
Visualize the DAG

### DISCUSSION THREAD

**Discussion by aluzzardi** [dagger/dagger#12055] [Ideas]
# Visualize the DAG

Being able to visualize the DAG, wether it is through an ASCII code or a web UI

_Originally posted by @grouville in https://github.com/dagger/dagger/discussions/2052#discussioncomment-2517752_

**Comment by 4x0v7** [comment]
Being able to diff the DAG would be awesome too, so you could understand how builds and caching layers might change.
A complex dagger plan visualised would really help with debugging and troubleshooting

**Comment by sword-jin** [comment]
Waiting this feature
